balloon.go

```diff
@@ -126,11 +129,15 @@ func (b *Instance) Mix(h hash.Hash, salt []byte, sCost, tCost uint64) {
 				h.Write(idxBlock)
 				otherBytes := h.Sum(nil)
 
-				for left, right := 0, len(otherBytes)-1; left < right; left, right = left+1, right-1 {
-					otherBytes[left], otherBytes[right] = otherBytes[right], otherBytes[left]
-				}
+				// math/big usually uses big-endian, with the exception of
+				// Int.SetBits. We can unsafely convert otherBytes directly
+				// into a big.Word slice to bypass the need to reverse
+				// otherBytes.
+				header := *(*reflect.SliceHeader)(unsafe.Pointer(&otherBytes))
```

Comment:
There should not be `reflect` and `unsafe` in crypto code. Also, it's hard to read (and to test).

Reply:
I did manage to convert it without `unsafe` in the original PR, not sure if it can be improved any further though.
```go
otherBytes := h.Sum(nil)
otherWords := make([]big.Word, h.Size() / (bits.UintSize / 8))


for byte, word := 0, 0; byte < len(otherBytes); byte, word = byte+bits.UintSize/8, word+1 {
	if bits.UintSize == 64 {
		otherWords[word] = big.Word(binary.LittleEndian.Uint64(otherBytes[byte : byte+bits.UintSize/8]))
	} else if bits.UintSize == 32 {
		otherWords[word] = big.Word(binary.LittleEndian.Uint32(otherBytes[byte : byte+bits.UintSize/8]))
	} else {
		panic("balloon: unsupported architecture")
	}
}


otherInt.SetBits(otherWords)
```
See https://github.com/nogoegst/balloon/pull/2/files#diff-43f7147f39b6ce50c058187412aeffffd6b79bf6c972dc475b1a86a674597901R128-R141.